interfaces/crypto/KeyPairInterface.h

```diff
@@ -0,0 +1,41 @@
+/**
+ *  Copyright (C) 2021 FISCO BCOS.
+ *  SPDX-License-Identifier: Apache-2.0
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ * @brief interfaces for KeyPair
+ * @file KeyPairInterface.h
+ * @author: yujiechen
+ * @date 2021-04-02
+ */
+#pragma once
+#include <bcos-framework/interfaces/crypto/Hash.h>
+#include <bcos-framework/interfaces/crypto/KeyInterface.h>
+namespace bcos
+{
+namespace crypto
+{
+class KeyPairInterface
+{
+public:
+    using Ptr = std::shared_ptr<KeyPairInterface>;
+    KeyPairInterface() = default;
+    virtual ~KeyPairInterface() {}
+
+    virtual SecretPtr secretKey() const = 0;
+    virtual PublicPtr publicKey() const = 0;
+    virtual PublicPtr priToPub(SecretPtr _secret) = 0;
```

Comment:
when KeyPair is constructed, the pub key and the private key is known, I think the `priToPub` is unnecessary